classes/course/BuildTermString.java

```diff
@@ -0,0 +1,83 @@
+public class BuildTermString {
+  public static String getTerm (String term) {
+    String result = "";
+
+    try {
+      switch(Integer.parseInt(term.substring(3,4))) {
+        case 2: result = "F";
+        break;
+        case 4: result = "W";
+        break;
+        case 6: result = "SP";
+        break;
+        case 8: result = "SU";
+        break;
+        default: result = "";
+        break;
+      }
+    }
+    catch (java.lang.StringIndexOutOfBoundsException e) {
+      System.out.println(e.getMessage());
+    }
+
+    return result;
+  }
+
+  public static String getLongTerm (String term) {
+    String result = "";
+
+    try {
+      switch(Integer.parseInt(term.substring(3,4))) {
+        case 2: result = "Fall";
+        break;
+        case 4: result = "Winter";
+        break;
+        case 6: result = "Spring";
+        break;
+        case 8: result = "Summer";
+        break;
+        default: result = "";
+        break;
+      }
+    }
+    catch (java.lang.StringIndexOutOfBoundsException e) {
+      System.out.println(e.getMessage());
+    }
+
+    return result;
+  }
+
+  public static String getYear (String term){
+    String result = "";
+    String century = "";
+    if(term.substring(0,1).equals("1")) {
+      century = "20";
+    }
+    String academicYear = term.substring(1,3);
+    int year = Integer.parseInt(academicYear);
+    if (getTerm(term).equals("F")) {
+      year = year - 1;
+    }
+
+    result = century + String.valueOf(year);
```

Comment:
The reserves app code is using a regex to strip out the first part of the class ID (see lines https://github.com/sul-dlss/course_reserves/blob/master/lib/course_work_courses.rb#L81 and https://github.com/sul-dlss/course_reserves/blob/master/lib/course_work_courses.rb#L84). It looks like we need to remove century from the result here.